### PR TITLE
Task/scaffolding app 13

### LINKED ISSUE #13: Scaffolding a flutter app
## Overview

Build a app scaffold with a basic infrastructure. Implement tooling which were decided in issue #12 

### Outcome 

Project in GitHub repository ready to be developed on and add new features on top of that

### Requirements

- routing prepared
- state management prepared
- database opening prepared
- splash screen prepared
- folder structure prepared
- basic services prepared
- database repositories prepared

## Task list

- [x] #23
- [x] #24
- [x] #26

### PR DESCRIPTION
**Pull request by storkandstars**
Closes #13

### CODE REVIEW

File: lib/view/widgets/project_toggle_item.dart [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  const ProjectToggleItem(
+      {Key? key, required this.project, this.toggled = false, this.onToggle})
+      : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _ProjectToggleItem();
+}
+
+class _ProjectToggleItem extends State<ProjectToggleItem> {
```
Comment (jvrastil) [inline lib/view/widgets/project_toggle_item.dart:17]: @storkandstars can you write some unit test for this class so the flow is fixed...?

**Review by jvrastil — APPROVED** [review]
looks good. If you have time please check that unit test

**Review by Juvi3eira — APPROVED** [review]
Looks good.

File: lib/view/widgets/project_toggle_item.dart (hunk 1 above)
Comment (jvrastil) [reply lib/view/widgets/project_toggle_item.dart:17]: I did it. @ciripel please review the changes

**Review by ciripel — APPROVED** [review]

**Pull request merged**